lib/runtime/invalid_transaction.go

```diff
@@ -0,0 +1,146 @@
+type Future struct{}
+
+// Index Returns VDT index
+func (Future) Index() uint { return 2 }
+
+// Stale General error to do with the transaction being outdated (e.g. nonce too low)
+type Stale struct{}
+
+// Index Returns VDT index
+func (Stale) Index() uint { return 3 }
+
+// BadProof General error to do with the transaction’s proofs (e.g. signature)
+type BadProof struct{}
+
+// Index Returns VDT index
+func (BadProof) Index() uint { return 4 }
+
+// AncientBirthBlock The transaction birth block is ancient
+type AncientBirthBlock struct{}
+
+// Index Returns VDT index
+func (AncientBirthBlock) Index() uint { return 5 }
+
+// ExhaustsResources The transaction would exhaust the resources of current block
+type ExhaustsResources struct{}
+
+// Index Returns VDT index
+func (ExhaustsResources) Index() uint { return 6 }
+
+var invalidCustom InvalidCustom
+
+// InvalidCustom Any other custom invalid validity that is not covered
+type InvalidCustom uint8
+
+// Index Returns VDT index
+func (InvalidCustom) Index() uint { return 7 }
+
+// BadMandatory An extrinsic with a Mandatory dispatch resulted in Error
+type BadMandatory struct{}
+
+// Index Returns VDT index
+func (BadMandatory) Index() uint { return 8 }
+
+// MandatoryDispatch A transaction with a mandatory dispatch
+type MandatoryDispatch struct{}
+
+// Index Returns VDT index
+func (MandatoryDispatch) Index() uint { return 9 }
+
+// Set will set a VaryingDataTypeValue using the underlying VaryingDataType
+func (i *InvalidTransaction) Set(val scale.VaryingDataTypeValue) (err error) {
+	vdt := scale.VaryingDataType(*i)
+	err = vdt.Set(val)
+	if err != nil {
+		return
+	}
+	*i = InvalidTransaction(vdt)
+	return
+}
+
+// Value will return value from underying VaryingDataType
+func (i *InvalidTransaction) Value() (val scale.VaryingDataTypeValue) {
+	vdt := scale.VaryingDataType(*i)
+	return vdt.Value()
+}
+
+// NewInvalidTransaction is constructor for InvalidTransaction
+func NewInvalidTransaction() InvalidTransaction {
+	vdt, err := scale.NewVaryingDataType(Call{}, Payment{}, Future{}, Stale{}, BadProof{}, AncientBirthBlock{},
+		ExhaustsResources{}, invalidCustom, BadMandatory{}, MandatoryDispatch{})
+	if err != nil {
+		panic(err)
+	}
+	return InvalidTransaction(vdt)
+}
+
+func (i *InvalidTransaction) Error() string {
+	switch val := i.Value().(type) {
+	case Call:
+		return errUnexpectedTxCall.Error()
+	case Payment:
+		return errInvalidPayment.Error()
+	case Future:
+		return errInvalidTransaction.Error()
+	case Stale:
+		return errOutdatedTransaction.Error()
+	case BadProof:
+		return errBadProof.Error()
+	case AncientBirthBlock:
+		return errAncientBirthBlock.Error()
+	case ExhaustsResources:
+		return errExhaustsResources.Error()
+	case InvalidCustom:
+		return newUnknownError(val).Error()
+	case BadMandatory:
+		return errMandatoryDispatchError.Error()
+	case MandatoryDispatch:
+		return errInvalidMandatoryDispatch.Error()
+	}
+
+	return errInvalidResult.Error()
```

Comment:
Should we panic instead here? Isn't this a programming error?

Reply:
hmm yes I suppose so. I basically am always scared to write panics in production code, but can add if best

Reply:
If it's a programming error and is not meant to happen, then panic ahead 👍 

Errors are for user/IO problems.